execution_pool.go

```diff
@@ -0,0 +1,36 @@
+package powermux
+
+import "sync"
+
+type executionPool struct {
+	p *sync.Pool
+}
+
+func (ep *executionPool) Get() *routeExecution {
+	return ep.p.Get().(*routeExecution)
+}
+
+func (ep *executionPool) Put(ex *routeExecution) {
+	ex.middleware = ex.middleware[0:0]
+	for key := range ex.params {
+		delete(ex.params, key)
+	}
+	ex.handler = nil
+	ex.notFound = nil
```

Comment:
I think this is too coupled with the `routeExecution` model. What about adding a function like `ex.Clean()`, so the routeExecution will know how to clean everything up? I'm just worried some day we add new attributes to the `routeExecution` struct and we forget to come here and handle it...